bundles/org.openhab.binding.mqtt.fpp/src/main/java/org/openhab/binding/mqtt/fpp/internal/handler/FPPPlayerHandler.java

```diff
@@ -0,0 +1,280 @@
+        bridgeStatusChanged(getBridgeStatus());
+    }
+
+    @Override
+    public void processMessage(String topic, byte[] payload) {
+        String state = new String(payload, StandardCharsets.UTF_8);
+        logger.trace("Received the following new FPP state:{}:{}", topic, state);
+
+        if (topic.endsWith(STATUS_TOPIC)) {
+            processIncomingState(state);
+        } else if (topic.endsWith(VERSION_TOPIC)) {
+            updateState(CHANNEL_VERSION, new StringType(state));
+            updateStatus(ThingStatus.ONLINE);
+        }
+    }
+
+    public ThingStatusInfo getBridgeStatus() {
+        Bridge b = getBridge();
+        if (b != null) {
+            return b.getStatusInfo();
+        } else {
+            return new ThingStatusInfo(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE, null);
+        }
+    }
+
+    @Override
+    public void bridgeStatusChanged(ThingStatusInfo bridgeStatusInfo) {
+        if (bridgeStatusInfo.getStatus() == ThingStatus.OFFLINE) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+            connection = null;
+            return;
+        }
+        if (bridgeStatusInfo.getStatus() != ThingStatus.ONLINE) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+            return;
+        }
+
+        Bridge localBridge = this.getBridge();
+        if (localBridge == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_UNINITIALIZED,
+                    "Bridge is missing or offline, you need to setup a working MQTT broker first.");
+            return;
+        }
+        ThingHandler handler = localBridge.getHandler();
+        if (handler instanceof AbstractBrokerHandler abh) {
+            final MqttBrokerConnection connection;
+            try {
+                connection = abh.getConnectionAsync().get(500, TimeUnit.MILLISECONDS);
+            } catch (InterruptedException | ExecutionException | TimeoutException ignored) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_UNINITIALIZED,
+                        "Bridge handler has no valid broker connection!");
+                return;
+            }
+            this.connection = connection;
+            updateStatus(ThingStatus.UNKNOWN, ThingStatusDetail.CONFIGURATION_PENDING,
+                    "Waiting for 'fpp/status: connected' MQTT message to be received. Check hub has 'MQTT Client Status Topic' configured.");
+            connection.subscribe(fullStatusTopic, this);
+            connection.subscribe(fullVersionTopic, this);
+        }
+        return;
+    }
+
+    private @Nullable String executeGet(String url) {
+        String response = null;
+        try {
+            response = HttpUtil.executeUrl("GET", "http://" + config.playerIP + url, 5000);
+
+        } catch (Exception e) {
+            logger.error("Failed HTTP Post", e);
```

Comment:
For reference: https://www.openhab.org/docs/developer/guidelines.html#f-logging
```suggestion
            logger.warn("Failed HTTP Post", e);
```